gnovm/pkg/gnolang/go2gno.go

```diff
@@ -469,6 +476,104 @@ func Go2Gno(fs *token.FileSet, gon ast.Node) (n Node) {
 	}
 }
 
+//----------------------------------------
+// type checking (using go/types)
+
+// MemPackageGetter implements the GetMemPackage() method. It is a subset of
+// [Store], separated for ease of testing.
+type MemPackageGetter interface {
+	GetMemPackage(path string) *std.MemPackage
+}
+
+// TypeCheckMemPackage performs type validation and checking on the given
+// mempkg. To retrieve dependencies, it uses getter.
+//
+// The syntax checking is performed entirely using Go's go/types package.
+func TypeCheckMemPackage(mempkg *std.MemPackage, getter MemPackageGetter) error {
+	var errs error
+	imp := &gnoImporter{
+		getter: getter,
+		cache:  map[string]gnoImporterResult{},
+		cfg: &types.Config{
+			Error: func(err error) {
+				errs = multierr.Append(errs, err)
+			},
+		},
+	}
+	imp.cfg.Importer = imp
+
+	_, err := imp.parseCheckMemPackage(mempkg)
+	// prefer to return errs instead of err:
+	// err will generally contain only the first error encountered.
+	if errs != nil {
+		return errs
+	}
+	return err
```

Comment:
nit: It might be clearer and probably less ambiguous to always return a multi-error here? 
```suggestion
	return multierr.Append(errs, err)
```

Reply:
See this: https://pkg.go.dev/go/types#Config.Error

Basically, the type checker allows returning multiple errors through this `Error` callback, which we specify above.

Even when you specify it, the error you get back from `types.Config.Check` is always going to be the first error encountered. So if we simply joined it with the existing `errs`, we would actually have the first error encountered duplicated in the error list

Additionally, the reason why we check the value of `err` at all is because the only case where it can be set (while `errs == nil`) is when parseCheckMemPackage returns an error before the final `Check` call